src/PageLifecycleProvider.tsx

```diff
@@ -66,6 +66,7 @@ export const PageLifecycleProvider: React.FC<PageLifecycleProviderProps> = ({
     const loadingDataCount = React.useRef(0)
     // Set to true when location changes, set to false again when update is complete
     const isRouting = React.useRef(true)
+    let started = false
```

Comment:
Is it possible to use a ref here, and leave the raise functions outside this function?

The value of start will be lost each render

Reply:
not sure how to use a ref here as it's immutable. I tried useState() but then the tests fail:
```
Warning: An update to PageLifecycleProvider inside a test was not wrapped in act(...).

      When testing, code that causes React state updates should be wrapped into act(...):

      act(() => {
        /* fire events that update state */
      });
      /* assert on the output */
```

Reply:
nevermind I was just using the wrong interface